### PR TITLE
Added max disk space go-diskqueue implementation

### PR DESCRIPTION
**Pull request by moredure**
Review please.

### CODE REVIEW

File: apps/nsqd/options.go [hunk 1]
```diff
@@ -148,7 +148,7 @@ func nsqdFlagSet(opts *nsqd.Options) *flag.FlagSet {
 	flagSet.String("data-path", opts.DataPath, "path to store disk-backed messages")
 	flagSet.Int64("mem-queue-size", opts.MemQueueSize, "number of messages to keep in memory (per topic/channel)")
 	flagSet.Int64("max-bytes-per-file", opts.MaxBytesPerFile, "number of bytes per diskqueue file before rolling")
-	flagSet.Int64("max-depth", opts.MaxDepth, "number of messages per diskqueue before queue stops accepting new messages")
+	flagSet.Int64("max-disk-space", opts.MaxDiskSpace, "max disk space of messages per diskqueue before queue starts removing old data")
```
Comment (telepenin) [inline apps/nsqd/options.go]: 1. Is it bytes, Mb, Kb?
2. is it specified per topic or for whole data, or per file?
3. how it removing the old data?

File: nsqd/nsqd_test.go [hunk 2]
```diff
@@ -157,6 +158,73 @@ func TestStartup(t *testing.T) {
...
 }
 
+func TestMaxDiskSpace(t *testing.T) {
+	iterations := 2048
+	doneExitChan := make(chan int)
+
+	opts := NewOptions()
+	opts.Logger = test.NewTestLogger(t)
+	opts.MemQueueSize = 0
+	opts.MaxBytesPerFile = 10240
```
Comment (telepenin) [inline nsqd/nsqd_test.go:169]: Do we need to specify "MaxBytesPerFile"?

File: nsqd/options.go [hunk 3]
```diff
@@ -120,7 +120,7 @@ func NewOptions() *Options {
 
 		MemQueueSize:    10000,
 		MaxBytesPerFile: 100 * 1024 * 1024,
-		MaxDepth:        0, //no max depth per queue
+		MaxDiskSpace:    0, //no max depth per queue
```
Comment (telepenin) [inline nsqd/options.go]: //no max depth per queue --> incorrect comment

File: nsqd/nsqd_test.go (hunk 2 above)
Comment (telepenin) [reply nsqd/nsqd_test.go:169]: When file starts to rollout?

File: apps/nsqd/options.go (hunk 1 above)
Comment (moredure) [reply apps/nsqd/options.go]: I'll rename to max-bytes-per-diskqueue or max-bytes-per-channel-and-topic ?

Comment (moredure) [reply apps/nsqd/options.go]: also we can separate to multiple different options max-bytes-per-channel/max-bytes-per-topic

File: nsqd/nsqd_test.go (hunk 2 above)
Comment (moredure) [reply nsqd/nsqd_test.go:169]: yes, in this test case, since by default maxBytesPerFile is much larger up to 100mb

File: nsqd/options.go (hunk 3 above)
Comment (moredure) [reply nsqd/options.go]: done

File: init.d/imunify-nsq.init.d [hunk 4]
```diff
@@ -43,7 +43,7 @@ start() {
         cd /
         close_fds
         export GOGC=10
-        setsid $exec --data-path /var/lib/imunify-nsqd --use-unix-sockets --tcp-address /var/run/imunify-nsqd.sock --http-address /var/run/imunify-nsqd-http.sock &
+        setsid $exec --data-path /var/lib/imunify-nsqd --max-bytes-per-queue 104857600 --use-unix-sockets --tcp-address /var/run/imunify-nsqd.sock --http-address /var/run/imunify-nsqd-http.sock &
```
Comment (telepenin) [inline init.d/imunify-nsq.init.d]: I suggest to use 1/10 for bytes per file to rollout to be sure that we removing only last 10% when it needed.

Comment (moredure) [reply init.d/imunify-nsq.init.d]: yeap, I'll set -max-bytes-per-file: 100mb total + 10mb per file?

Comment (telepenin) [reply init.d/imunify-nsq.init.d]: ok

**Review by telepenin — APPROVED** [review]

**Pull request merged**